source/configuration_manual/forwarding_parameters.rst

```diff
@@ -1,74 +1,85 @@
 
 Supported parameters
-====================
+~~~~~~~~~~~~~~~~~~~~
 
-* ``x-originating-ip`` - Client IP
+-  ``x-originating-ip`` - Client IP
 
-* ``x-originating-port`` - Client port
+-  ``x-originating-port`` - Client port
 
-* ``x-connected-ip`` - Server IP
+-  ``x-connected-ip`` - Server IP
 
-* ``x-connected-port`` - Server port
+-  ``x-connected-port`` - Server port
 
-* ``x-proxy-ttl`` - TTL which is reduced by each hop, loop prevention. When TTL drops to 0, the connection is dropped.
+-  ``x-proxy-ttl`` - TTL which is reduced by each hop, loop prevention.
+   When TTL drops to 0, the connection is dropped.
 
-* ``x-session-id / x-session-ext-id`` - Session ID to be used.
+-  ``x-session-id`` / ``x-session-ext-id`` - Session ID to be used.
 
-* ``x-forward-<variable_name>``- Forwarded variable, see :ref:`config_variables`
+-  ``x-forward-<variable_name>`` - Forwarded variable, see
+   :ref:`Variables <config_variables>`
 
 POP3 protocol
-=============
+-------------
 
-For POP3 protocol, this is done with custom ``XCLIENT`` command which accepts a space separated list of ``key=value`` parameters.
+For POP3 protocol, this is done with custom ``XCLIENT`` command which
+accepts a space separated list of key=value parameters.
 
 Supported parameters
-====================
+~~~~~~~~~~~~~~~~~~~~
 
-* ``ADDR`` - Client IP
+-  ``ADDR`` - Client IP
 
-* ``PORT`` - Client port
+-  ``PORT`` - Client port
 
-* ``SESSION`` - Session ID
+-  ``SESSION`` - Session ID
 
-* ``TTL`` - TTL which is reduced by each hop, loop prevention. When TTL drops to 0, the connection is dropped.
+-  ``TTL`` - TTL which is reduced by each hop, loop prevention. When TTL
+   drops to 0, the connection is dropped.
 
-* ``FORWARD`` - Base64 encoded ``key=value`` parameter to be stored.
+-  ``FORWARD`` - Base64 encoded key=value parameter to be stored.
 
 SMTP/LMTP protocol
-==================
+------------------
 
 See http://www.postfix.org/XCLIENT_README.html
 
-* ``ADDR`` - Client IP, IPV6: prefix is needed for IPv6.
+-  ``ADDR`` - Client IP, IPV6: prefix is needed for IPv6.
 
-* ``PORT`` - Client port
+-  ``PORT`` - Client port
 
-* ``SESSION`` - Session ID
+-  ``SESSION`` - Session ID
 
-* ``HELO`` - Original HELO/EHLO
+-  ``HELO`` - Original HELO/EHLO
 
-* ``LOGIN`` - Original LOGIN value
+-  ``LOGIN`` - Original LOGIN value
 
-* ``TIMEOUT`` - Original TIMEOUT
+-  ``TIMEOUT`` - Original TIMEOUT
 
-* ``TTL`` - TTL which is reduced by each hop, loop prevention. When TTL drops to 0, the connection is dropped.
+-  ``TTL`` - TTL which is reduced by each hop, loop prevention. When TTL
+   drops to 0, the connection is dropped.
 
-* ``PROTO`` - Forwarded protocol, can be one of SMTP, ESTMP, LMTP.
+-  ``PROTO`` - Forwarded protocol, can be one of SMTP, ESTMP, LMTP.
 
-* ``FORWARD`` - Base64 encoded key=value parameter to be stored.
\ No newline at end of file
+-  ``FORWARD`` - Base64 encoded key=value parameter to be stored.
```

Comment:
This XCLIENT FORWARD exists specifically for Submission, not for LMTP. The total XCLIENT command line length is limited to 1033 bytes (not sure why).

LMTP implements RCPT TO .. XRCPTFORWARD parameter. This is the same key=value tab-escaped field as with POP3. This is effectively limited to around 900 bytes, because larger values result in:

```
Sep 28 00:48:08.816376 auth: Error: auth-master client: BUG: Master sent us more than 1024 bytes (created 0 msecs ago, handshake 0 msecs ago)
Sep 28 00:48:08.816549 lmtp(895928): Error: auth-master: passdb lookup(tss): Disconnected unexpectedly
```

This happens only for trusted clients, so not a problem for outside connections.

Reply:
Wasn't quite sure what info was ok for the page, so check my new edits.